src/certificates/certificate-manager/CertificateManager.ts

```diff
@@ -0,0 +1,106 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-expect-error
+import rs from "jsrsasign";
+
+export interface CertificatePem {
+  certKey: string;
+  publicKey: string;
+  privateKey: string;
+}
+
+export interface CertificateInfo {
+  hSerial: string;
+  sIssuer: string;
+  sSubject: string;
+  sNotBefore: string;
+  sNotAfter: string;
+  issuedOn: Date;
+  expiresOn: Date;
+}
+
+export class CertificateManager {
+  parsePem(certPEM: string): CertificateInfo {
+    const certificate = new rs.X509();
+    certificate.readCertPEM(certPEM);
+    const hSerial: string = certificate.getSerialNumberHex();
+    const sIssuer: string = certificate.getIssuerString();
+    const sSubject: string = certificate.getSubjectString();
+    const sNotBefore: string = certificate.getNotBefore();
+    const sNotAfter: string = certificate.getNotAfter();
+
+    return {
+      hSerial,
+      sIssuer,
+      sSubject,
+      sNotBefore,
+      sNotAfter,
+      issuedOn: this.strToDate(sNotBefore),
+      expiresOn: this.strToDate(sNotAfter)
+    };
+  }
+
+  generatePEM(address: string): CertificatePem {
+    const { notBeforeStr, notAfterStr } = this.createValidityRange();
+    const { prvKeyObj, pubKeyObj } = rs.KEYUTIL.generateKeypair("EC", "secp256r1");
+    const cert = new rs.KJUR.asn1.x509.Certificate({
+      version: 3,
+      serial: { int: Math.floor(new Date().getTime() * 1000) },
+      issuer: { str: "/CN=" + address },
+      notbefore: notBeforeStr,
+      notafter: notAfterStr,
```

Comment:
No need to add right now, but it would be nice to be able to specify the expiry date when creating a cert.

Reply:
not a big deal, here u are https://github.com/akash-network/akashjs/pull/89/files#diff-dda604b0726bf284b6caf45c9edcb587ed8b9ce1c1cda250de7a39aed1eea1cbR79